microsoft/terminal
# About Dialog OS Version

## Summary of the Pull Request
Adds the user's full Windows version to the About dialog page. This reduces the amount of work a user must perform to open an Issue.

 
## References
https://docs.microsoft.com/en-us/windows/win32/sysinfo/getting-the-system-version


## PR Checklist
* [x] Closes #3616
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #3616


## Detailed Description of the Pull Request / Additional comments
PR adds a function `_GetWindowVersion` to `TerminalPage` which queries the version of an important Sytem32 DLL. The About dialog page uses this function to query the system version. While the referenced link on getting the Windows version suggests querying `kernel32.dll`, I found its version number was inconsistent compared to the version given when `ver` is run in a `cmd` terminal. This implementation queries `KernelBase.dll` instead which gives the correct version.

PR also changes the Windows Terminal version line to read `WT Version: x.x.x.x` to clearly differentiate the OS and WT versions.


## Validation Steps Performed
To validate the correctness of the About dialog OS version:
- Run the command `ver` in a `cmd` terminal
- Open the About dialog page in WT
- Compare the two Windows version numbers

To test error checking and case where DLL does not exist, I replaced `KernelBase.dll` with `does_not_exist.dll` and re-ran WT and got the expected `OS Version: 0.0.0.0 <arch>`.

<img width="383" alt="ver-number" src="https://user-images.githubusercontent.com/28279285/69572005-1588cb00-0f78-11ea-8041-28807f396c91.PNG">
<img width="309" alt="in-progress" src="https://user-images.githubusercontent.com/28279285/69578973-69021580-0f86-11ea-92f9-bfd326d6d6d2.PNG">
